src/composables/deeplink-api.ts

```diff
@@ -0,0 +1,33 @@
+import { ref } from '@vue/composition-api';
+import VueRouter, { Route } from 'vue-router';
+
+export interface UseDeepLinkApiOptions {
+  router: VueRouter
+}
+
+export function useDeepLinkApi({ router }: UseDeepLinkApiOptions) {
```

Comment:
I'm not 100% sure, but I think there's no need to pass the router as an argument. You can import the router instance from src/popup/router/index.js

Reply:
That will cause a Dependency import cycle, once we upgrade `vue-router: 4.x.x`we can replace It with `useRouter`

Reply:
Tested again and you have right. There is no simple way to avoid the circular dependencies in the page components. I suggest only to add a comment why we need to pass the router to make the work easier for other devs.